File: doc/MEDMEM/MEDMEM_InvokingDriverByAttachingItToAnObject.cxx
//  Copyright (C) 2007-2010  CEA/DEN, EDF R&D, OPEN CASCADE
//
//  Copyright (C) 2003-2007  OPEN CASCADE, EADS/CCR, LIP6, CEA/DEN,
//  CEDRAT, EDF R&D, LEG, PRINCIPIA R&D, BUREAU VERITAS
//
//  This library is free software; you can redistribute it and/or
//  modify it under the terms of the GNU Lesser General Public
//  License as published by the Free Software Foundation; either
//  version 2.1 of the License.
//
//  This library is distributed in the hope that it will be useful,
//  but WITHOUT ANY WARRANTY; without even the implied warranty of
//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
//  Lesser General Public License for more details.
//
//  You should have received a copy of the GNU Lesser General Public
//  License along with this library; if not, write to the Free Software
//  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
//
//  See http://www.salome-platform.org/ or email : webmaster.salome@opencascade.com
//

#include "MEDMEM_Exception.hxx"
#include "MEDMEM_define.hxx"

#include "MEDMEM_Field.hxx"
#include "MEDMEM_Mesh.hxx"
#include "MEDMEM_Med.hxx"
#include "MEDMEM_MedMedDriver.hxx"
#include "MEDMEM_MedMeshDriver.hxx"

using namespace MEDMEM ;
using namespace MED_EN ;

main () {
  
  const char * fileName   = "pointe.med";
  const char * fileName2  = "Field&MeshGeneratedPointe.med";
  const char * fileName3  = "MedGeneratedPointe.med";
  const char * fieldName  = "fieldcelldouble";
  const char * meshName   = "maa1";

  try {
    // Test creation of drivers from the standard driver method of an object
    {  
      FIELD<double> * myField = new FIELD<double>();
      MED_FIELD_RDONLY_DRIVER<double> myRdOnlyDriver(fileName,myField);
      myRdOnlyDriver.setFieldName(fieldName);
      myRdOnlyDriver.open(); 
      //This test failed due to inadequate Support implementation   
      // myRdOnlyDriver.read();
      // try  { myRdOnlyDriver.write(); } catch  (MEDEXCEPTION& ex) 
      //   { MESSAGE(ex.what()); }
      MED_FIELD_WRONLY_DRIVER<double> myWrOnlyDriver(fileName2,myField);
      myWrOnlyDriver.open(); 
      //This test failed due to inadequate Support implementation   
      // myWrOnlyDriver.write(); 
      // try  myWrOnlyDriver.read(); catch  (MEDEXCEPTION& ex) 
      //   { MESSAGE(ex.what()); }
      myRdOnlyDriver.close();
      myWrOnlyDriver.close();
      delete myField;
    }

    {
      MESH * myMesh = new MESH();
      MED_MESH_RDONLY_DRIVER myRdOnlyDriver(fileName,myMesh);
      myRdOnlyDriver.setMeshName(meshName);
      myRdOnlyDriver.open(); 
      myRdOnlyDriver.read();
      myRdOnlyDriver.close(); 
      // try  { myRdOnlyDriver.write(); } catch  (MEDEXCEPTION& ex)
      //   { MESSAGE(ex.what()); }
      MED_MESH_WRONLY_DRIVER myWrOnlyDriver(fileName2,myMesh);
      myWrOnlyDriver.setMeshName(meshName);
      myWrOnlyDriver.open(); 
      myWrOnlyDriver.write(); 
      // try  myWrOnlyDriver.read(); catch  (MEDEXCEPTION& ex)
      //   { MESSAGE(ex.what()); }
      // myRdOnlyDriver.close(); 
      //While we use H5close() in the MESH/FIELD drivers, the next
      //line will fail, because all files are previously closed !
      myWrOnlyDriver.close();
      delete myMesh;
    }

    {
      MED * myMed = new MED();
      MED_MED_RDONLY_DRIVER myRdOnlyDriver(fileName,myMed);
      myRdOnlyDriver.open(); 
      myRdOnlyDriver.readFileStruct();
      myRdOnlyDriver.close(); 
      myMed->updateSupport(); // DOIT ETRE SUPPRIMEE
      //      myRdOnlyDriver.read();
      // try { myRdOnlyDriver.write(); } catch  (MEDEXCEPTION& ex) 
      //   { MESSAGE(ex.what()); }
      //MED_MED_WRONLY_DRIVER myWrOnlyDriver(fileName3,myMed);
      //myWrOnlyDriver.open(); 
      //myWrOnlyDriver.write(); // Not implemented yet.
      //myWrOnlyDriver.close();
      delete myMed;
    }

  } catch (MEDEXCEPTION& ex){
    cout << "MAIN BLOCK EXCEPTION" << endl;
    MESSAGE(ex.what()) ;
  }
}

File: doc/MEDMEM/MEDMEM_InvokingDriverByAttachingItToAnObject.py
#  -*- coding: iso-8859-1 -*-
#  Copyright (C) 2007-2010  CEA/DEN, EDF R&D, OPEN CASCADE
#
#  Copyright (C) 2003-2007  OPEN CASCADE, EADS/CCR, LIP6, CEA/DEN,
#  CEDRAT, EDF R&D, LEG, PRINCIPIA R&D, BUREAU VERITAS
#
#  This library is free software; you can redistribute it and/or
#  modify it under the terms of the GNU Lesser General Public
#  License as published by the Free Software Foundation; either
#  version 2.1 of the License.
#
#  This library is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
#  Lesser General Public License for more details.
#
#  You should have received a copy of the GNU Lesser General Public
#  License along with this library; if not, write to the Free Software
#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
#
#  See http://www.salome-platform.org/ or email : webmaster.salome@opencascade.com
#

######################################################################
# This Python script should be executed when the shared library is   #
# generated using SWIG 1.3 (or higher) due to the fact that older    #
# version could not handle the wrapping of several class constructor #
######################################################################
#
from libMEDMEM_Swig import *

medFile = "pointe.med"
medFile2 = "Field&MeshGeneratedPointe.med"
fieldName = "fieldcelldoublescalar"
meshName = "maa1"

try:
    myField = FIELDDOUBLE()
    myRdOnlyDriver = MED_FIELDDOUBLE_RDONLY_DRIVER(medFile,myField)
    myRdOnlyDriver.setFieldName(fieldName)
    myRdOnlyDriver.open()

    myWrOnlyDriver = MED_FIELDDOUBLE_WRONLY_DRIVER(medFile2,myField)
    myWrOnlyDriver.open()

    myRdOnlyDriver.close()
    myWrOnlyDriver.close()

    print "Invoking field drivers OK"
except :
    print "there is a problem in invoking field drivers !!"
    print "Please consult the error standart output of the python execution !!"

try:
    myMesh = MESH()

    myRdOnlyDriver = MED_MESH_RDONLY_DRIVER(medFile,myMesh)
    myRdOnlyDriver.setMeshName(meshName)
    myRdOnlyDriver.open()
    myRdOnlyDriver.read()
    myRdOnlyDriver.close()

    myWrOnlyDriver = MED_MESH_WRONLY_DRIVER(medFile2,myMesh)
    myWrOnlyDriver.setMeshName(meshName)
    myWrOnlyDriver.open()
    myWrOnlyDriver.write()
    myWrOnlyDriver.close()

    print "Invoking mesh drivers OK"
except :
    print "there is a problem in invoking mesh drivers !!"
    print "Please consult the error standart output of the python execution !!"

